pkg/controller/migration/convert/core.go

```diff
@@ -217,7 +218,8 @@ func addResources(install *operatorv1.Installation, compName operatorv1.Componen
 // since Operator does not support setting custom annotations on components, these annotations
 // would otherwise be dropped.
 func handleAnnotations(c *components, _ *operatorv1.Installation) error {
-	if a := removeExpectedAnnotations(c.node.Annotations, map[string]string{}); len(a) != 0 {
+	kubectlGenerated := []string{"^kubectl\\.kubernetes\\.io"}
```

Comment:
Minor: Might be more appropriate to have a generic name for this (e.g. regexIgnoreList). Since kubectlGenerated is just one specific example of an expected pattern that we want to remove from the annotations.